### PR TITLE
Homework 2

### PR DESCRIPTION
**Pull request by povilassta**
Added passport-jwt strategy, mock data for users to test authentication, route protection.

### CODE REVIEW

File: services/auth.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+import jwt from "jsonwebtoken";
+
+// Jwt strategy
+const opts = {
+  jwtFromRequest: ExtractJwt.fromAuthHeaderAsBearerToken(),
+  secretOrKey: process.env.JWT_SECRET,
+};
+
+const jwtStrategy = new JwtStrategy(opts, async (payload, done) => {
+  const user = db.getUserByEmail(payload.email);
```
Comment (Lokki17) [inline services/auth.js:15]: I think db.getUserByEmail(payload.email) should be async func (maybe not now but in future)

File: services/auth.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  if (user) {
+    return done(null, user);
+  }
+  return done(null, false);
```
Comment (Lokki17) [inline services/auth.js:19]: return done(null, user || false) looks better :)

File: services/auth.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  } else {
+    // If user doesn't exist or password is incorrect
+    res.status(401).json({ message: "Invalid credentials" });
+  }
+}
+
+export function register(req, res) {
+  const { email, password, role } = req.body;
+  const user = db.registerUser({ email, password, role });
+  const { password: pass, ...rest } = user;
```
Comment (Lokki17) [inline services/auth.js:58]: Hmm... I dont see rest variable. Does this code work?

**Review by Lokki17 — CHANGES_REQUESTED** [review]
Fix remarks and merge

File: services/auth.js (hunk 3 above)
Comment (povilassta) [reply services/auth.js:58]: image
Yes, it returns json like this.

**Pull request merged**